ext/ensemble_builder.jl

```diff
@@ -205,123 +243,27 @@ function _try_fill_g_ens_col_with_var!(
+    var::OutputVar,
+    metadata::Metadata;
+    verbose = false,
 )
-
-"""
-    _same_units(var::OutputVar, metadata::Metadata)
-
-Return `true` if `var` and `metadata` have the same units, `false` otherwise.
-"""
-function _same_units(var::OutputVar, metadata::Metadata)
-    var_units = ClimaAnalysis.units(var)
-    metadata_units = ClimaAnalysis.units(metadata)
-    if var_units == "" || metadata_units == ""
-        @warn("Var or metadata may be missing units")
-    end
-    return var_units == metadata_units
-end
-
-"""
-    _same_dim_units(var::OutputVar, metadata::Metadata)
-
-Return `true` if the units of the dimensions in `var` and `metadata` are the
-same, `false` otherwise. This function assumes `var` and `metadata` have the
-same dimensions.
-"""
-function _same_dim_units(var::OutputVar, metadata::Metadata)
-    for var_dim_name in keys(var.dims)
-        md_dim_name = ClimaAnalysis.Var.find_corresponding_dim_name_in_var(
-            var_dim_name,
-            metadata,
-        )
-        var_dim_units = ClimaAnalysis.dim_units(var, var_dim_name)
-        md_dim_units = ClimaAnalysis.dim_units(metadata, md_dim_name)
-        var_dim_units == md_dim_units || return false
-        if var_dim_units == "" || md_dim_units == ""
-            @warn(
-                "Units for $(ClimaAnalysis.conventional_dim_name(var_dim_name)) is missing in var or metadata"
-            )
-        end
-    end
-    return true
-end
-
-"""
-    _compatible_dims_values(var::OutputVar, metadata::Metadata)
-
-Check if the values of the dimensions in `var` and `metadata` are compatible
-for the purpose of filling out the G ensemble matrix.
-
-The nontemporal dimensions are compatible if the values are approximately the
-same. The temporal dimensions are compatible if the temporal dimension of
-`metadata` is a subset of the temporal dimension of `var`.
-"""
-function _compatible_dims_values(var::OutputVar, metadata::Metadata)
-    for var_dim_name in keys(var.dims)
-        md_dim_name = ClimaAnalysis.Var.find_corresponding_dim_name_in_var(
-            var_dim_name,
-            metadata,
-        )
-        if ClimaAnalysis.conventional_dim_name(var_dim_name) != "time"
-            all(isapprox(var.dims[var_dim_name], metadata.dims[md_dim_name])) ||
-                return false
-        else
-            dates_or_times(metadata) ⊆ dates_or_times(var) || return false
-        end
-    end
-    return true
-end
-
-"""
-    dates_or_times(var::OutputVar)
-
-Return the temporal dimension of `var` as dates if possible and time otherwise.
-
-This function assumes that a temporal dimension exists in `var`.
-"""
-function dates_or_times(var_or_metadata::Union{OutputVar, Metadata})
-    temporal_dim = try
-        ClimaAnalysis.dates(var_or_metadata)
-    catch
-        ClimaAnalysis.times(var_or_metadata)
-    end
-    return temporal_dim
+    return all(
+        Checker.check(checker, var, metadata; verbose = verbose) for
+        checker in checkers
+    )
```

Comment:
```suggestion
    for checker in checkers
        Checker.check(checker, var, metadata; verbose = verbose) || return false
    end
    return true
```

Minor, but this could exit early when there is a false check

Reply:
I think `all` does an early exit already.

```julia
julia> function foo(x)
       x == 1 && false
       x == 2 && error("foo")
       end
foo (generic function with 1 method)

julia> all(foo(x) for x in 1:2)
false

julia> all(foo(x) == false for x in 1:2)
ERROR: foo
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] foo
   @ ./REPL[14]:3 [inlined]
 [3] #9
   @ ./none:0 [inlined]
 [4] iterate
   @ ./generator.jl:48 [inlined]
 [5] _all
   @ ./reduce.jl:1316 [inlined]
 [6] all
   @ ./reduce.jl:1301 [inlined]
 [7] all(itr::Base.Generator{UnitRange{Int64}, var"#9#10"})
   @ Base ./reduce.jl:1193
 [8] top-level scope
   @ REPL[16]:1
 [9] top-level scope
   @ none:1
```

Reply:
oh great, then this is unnecessary